bin/setup-ssh-via-ssm.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+#
+# This script uploads your public SSH key ('id_ed25519' by default)
+# to a home directory ('ec2-user' by default) on an EC2 instance.
+# Written by Bruno Grande.
+#
+# Usage:    bash setup-ssh-via-ssm.sh INSTANCE_ID
+#
+# Example:  bash setup-ssh-via-ssm.sh i-0d3ff331a7b7a7e8b
+#
+#
+# This script assumes that:
+#
+# - You have generated an ed25519 SSH key pair as per
+#   GitHub's best practices. Otherwise, you'll need to
+#   generate one using:
+#
+#     ssh-keygen -t ed25519 -C "your_email@example.com"
+#
+# - Your shell environment is configured with AWS credentials
+#   (AWS_PROFILE or AWS_ACCESS_KEY_ID/AWS_SECRET_ACCESS_KEY).
+#   For example, if you have an AWS CLI profile based on the
+#   Service Catalog docs, you will first need to run this:
+#
+#     export AWS_PROFILE=service-catalog
+#
+#   Tip: If you only use the AWS CLI for the Service Catalog,
+#   consider making the config profile the default as follows:
+#
+#     [default]
+#     region=us-east-1
+#     credential_process = ...
+#
+# - Your SSH configuration file includes the following snippet
+#   (update the value for 'IdentityFile' as appropriate):
+#
+#     Host i-* mi-*
+#         ProxyCommand sh -c "aws ssm start-session --target %h --document-name AWS-StartSSHSession --parameters 'portNumber=%p'"
+#         IdentityFile ~/.ssh/id_ed25519
+#         User ec2-user
+#
+
+# Command-line arguments and optional environment variables
+INSTANCE_ID="${1}"
+SSH_KEY_PUB="${SSH_KEY_PUB:-$HOME/.ssh/id_ed25519.pub}"
```

Comment:
nit: I have multiple SSH keys, so I might want to specify this without editing the file

Reply:
The public SSH key is now an optional positional argument!